### PR TITLE
Have downloads always end up under WORKING_DIRECTORY

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #5912

### CODE REVIEW

File: pulpcore/download/base.py [hunk 1]
```diff
@@ -127,7 +128,12 @@ def _ensure_writer_has_open_file(self):
             # write the file to the current working directory with a random prefix and the
             # desired suffix. we always want the random prefix as it is possible to download
             # the same filename from two different URLs, and the files may not be the same.
-            self._writer = tempfile.NamedTemporaryFile(dir=".", suffix=suffix, delete=False)
+            work_dir = str(settings.WORKING_DIRECTORY)
+            self._writer = tempfile.NamedTemporaryFile(
+                dir="." if work_dir in os.getcwd() else work_dir,
+                suffix=suffix,
+                delete=False,
+            )
```
Comment (lubosmj) [inline pulpcore/download/base.py:136]: It appears like this is a better variant of what I was trying to do in the past: https://github.com/pulp/pulp_container/pull/1299#discussion_r1291196612.

Comment (gerrod3) [reply pulpcore/download/base.py:136]: Yeah, I'm doing something similar with my new work on pull-through caching in pulp-python and I thought I should just move the fix into pulpcore.

Comment (mdellweg) [reply pulpcore/download/base.py:136]: Hmm this feels like patching over a deeper issue. But well.
I think we should leave a comment on why this is important and the assumptions that lead to there.

File: pulpcore/download/base.py [hunk 2]
```diff
@@ -127,7 +128,12 @@ def _ensure_writer_has_open_file(self):
             # write the file to the current working directory with a random prefix and the
             # desired suffix. we always want the random prefix as it is possible to download
             # the same filename from two different URLs, and the files may not be the same.
-            self._writer = tempfile.NamedTemporaryFile(dir=".", suffix=suffix, delete=False)
+            work_dir = str(settings.WORKING_DIRECTORY)
+            self._writer = tempfile.NamedTemporaryFile(
+                dir="." if work_dir in os.getcwd() else work_dir,
```
Comment (mdellweg) [inline pulpcore/download/base.py:133]: This is a simple string comparison. So at least it should be `os.getcwd().startswith(work_dir)`. But i think i'd prefer using pathlib at this point.

Comment (gerrod3) [reply pulpcore/download/base.py]: Ok, updated to use pathlib logic and added a simple comment for the check.

**Review by mdellweg — APPROVED** [review]

**Pull request merged**